app/models/glitch/keyword_mute.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+# == Schema Information
+#
+# Table name: glitch_keyword_mutes
+#
+#  id         :integer          not null, primary key
+#  account_id :integer          not null
+#  keyword    :string           not null
+#  whole_word :boolean          default(TRUE), not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Glitch::KeywordMute < ApplicationRecord
+  belongs_to :account, required: true
+
+  validates_presence_of :keyword
+
+  after_commit :invalidate_cached_matcher
+
+  def self.matcher_for(account_id)
+    Matcher.new(account_id)
+  end
+
+  private
+
+  def invalidate_cached_matcher
+    Rails.cache.delete("keyword_mutes:regex:#{account_id}")
+  end
+
+  class Matcher
+    attr_reader :account_id
+    attr_reader :regex
+
+    def initialize(account_id)
+      @account_id = account_id
+      @regex = Rails.cache.fetch("keyword_mutes:regex:#{account_id}") { regex_for_account }
```

Comment:
Just realized that this will still cache the `Regexp` object, not the string describing the regex.  I'm not sure how that plays out with e.g. changing Ruby versions as part of a Mastodon upgrade (which will happen, and such an upgrade isn't guaranteed to clear the cache).  I'll modify this to only cache the string, which I think raises fewer questions.

Reply:
Cached object changed to the regex text in 8410d33b49d66683f5765b6c6ee5a4a1af5b098f.